### DISCUSSION TITLE
Create dtls server accepting multiple clients - Windows 11

### DISCUSSION THREAD

**Discussion by GurixD** [openssl/openssl#24588] [Q&A]
Windows 11
OpenSSL 3.0.3

I'm unable to create a dtls server accepting multiple clients creating a native socket for each. It appears that all the messages sent by clients are always received only by the mainfd and never by the peerfd making the SSL_accept hang forever. This makes me think about https://github.com/openssl/openssl/issues/6934#issuecomment-428198714 because closing the mainfd will make the messages receivable by peerfd. (tried with dtls_udp_echo.c program which works fine on linux)

I tried with bare udp sockets to do the same and only the mainfd is receiving on windows while on linux only peerfd is receiving:
https://gist.github.com/GurixD/becb340fcee09ce2fec24718373688eb 

How can I accept multiple clients on windows?

**Comment by alorbach** [comment]
Damn I run into the same problem now. It appears to be a limitation because Windows is not supporting SO_REUSEPORT in setsockopt but only SO_REUSEADDR. I have not found a workaround until which is really devastating and preventing me from creating a full multiclient DTLS Server.

Have you found any solution to this problem ?

**Comment by alorbach** [comment]
For your information, I have written a DTLS Server as plugin for rsyslog which supports multiple clients using SO_REUSEADDR and SO_REUSEPORT to run multiple sockets on the same port / address. But this only works on Linux so far, as Windows is missing support for SO_REUSEPORT:

https://github.com/rsyslog/rsyslog/blob/master/plugins/imdtls/imdtls.c
